auth0/resource_auth0_log_stream.go

```diff
@@ -0,0 +1,346 @@
+	}
+
+	d.SetId(auth0.StringValue(c.ID))
+	d.Set("name", c.Name)
+	d.Set("status", c.Status)
+	d.Set("type", c.Type)
+	d.Set("sink", flattenLogStreamSink(d, c.Sink))
+	return nil
+}
+
+func updateLogStream(d *schema.ResourceData, m interface{}) error {
+	c := expandLogStream(d)
+	api := m.(*management.Management)
+	err := api.LogStream.Update(d.Id(), c)
+	if err != nil {
+		return err
+	}
+	return readLogStream(d, m)
+}
+
+func deleteLogStream(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	err := api.LogStream.Delete(d.Id())
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok {
+			if mErr.Status() == http.StatusNotFound {
+				d.SetId("")
+				return nil
+			}
+		}
+	}
+	return err
+}
+
+func flattenLogStreamSink(d ResourceData, sink interface{}) []interface{} {
+
+	var m interface{}
+
+	switch o := sink.(type) {
+	case *management.LogStreamSinkAmazonEventBridge:
+		m = flattenLogStreamSinkAmazonEventBridge(o)
+	case *management.LogStreamSinkAzureEventGrid:
+		m = flattenLogStreamSinkAzureEventGrid(o)
+	case *management.LogStreamSinkHTTP:
+		m = flattenLogStreamSinkHTTP(o)
+	case *management.LogStreamSinkDatadog:
+		m = flattenLogStreamSinkDatadog(o)
+	case *management.LogStreamSinkSplunk:
+		m = flattenLogStreamSinkSplunk(o)
+	}
+	return []interface{}{m}
+}
+
+func flattenLogStreamSinkAmazonEventBridge(o *management.LogStreamSinkAmazonEventBridge) interface{} {
+	return map[string]interface{}{
+		"aws_account_id":           o.GetAccountID(),
+		"aws_region":               o.GetRegion(),
+		"aws_partner_event_source": o.GetPartnerEventSource(),
+	}
+}
+
+func flattenLogStreamSinkAzureEventGrid(o *management.LogStreamSinkAzureEventGrid) interface{} {
+	return map[string]interface{}{
+		"azure_subscription_id": o.GetSubscriptionID(),
+		"azure_resource_group":  o.GetResourceGroup(),
+		"azure_region":          o.GetRegion(),
+		"azure_partner_topic":   o.GetPartnerTopic(),
+	}
+}
+
+func flattenLogStreamSinkHTTP(o *management.LogStreamSinkHTTP) interface{} {
+	return map[string]interface{}{
+		"http_endpoint":       o.GetEndpoint(),
+		"http_contentFormat":  o.GetContentFormat(),
+		"http_contentType":    o.GetContentType(),
+		"http_authorization":  o.GetAuthorization(),
+		"http_custom_headers": o.CustomHeaders,
+	}
+}
+
+func flattenLogStreamSinkDatadog(o *management.LogStreamSinkDatadog) interface{} {
+	return map[string]interface{}{
+		"datadog_region":  o.GetRegion(),
+		"datadog_api_key": o.GetAPIKey(),
+	}
+}
+
+func flattenLogStreamSinkSplunk(o *management.LogStreamSinkSplunk) interface{} {
+	return map[string]interface{}{
+		"splunk_domain": o.GetDomain(),
+		"splunk_token":  o.GetToken(),
+		"splunk_port":   o.GetPort(),
+		"splunk_secure": o.GetSecure(),
+	}
+}
+func expandLogStream(d ResourceData) *management.LogStream {
+
+	ls := &management.LogStream{
+		Name:   String(d, "name", IsNewResource()),
+		Type:   String(d, "type", IsNewResource()),
+		Status: String(d, "status"),
```

Comment:
The `name` field is allowed to update apparently. See the [api docs](https://auth0.com/docs/api/management/v2#!/Log_Streams/patch_log_streams_by_id).

The `status` is a first for this provider. It can only be updated, not created. See also the suggestion on `createLogStream`.

```suggestion
		Name:   String(d, "name"),
		Type:   String(d, "type", IsNewResource()),
		Status: String(d, "status", Not(IsNewResource())),
```

Reply:
`Not` doesn't yet exist. But it's a trivial condition we can add to `resource_data.go`

```go
// Not is a condition that evaluates to true if its child condition evaluates to
// false. False otherwise.
func Not(condition Condition) Condition {
	return func(d ResourceData, key string) bool {
		return !condition.Eval(d, key)
	}
}
```